src/share/constants.hpp

```diff
@@ -23,7 +23,7 @@ class constants final {
     return "/Library/Application Support/org.pqrs/tmp/karabiner_event_dispatcher_receiver";
   }
 
-  static const char* get_home_dot_karabiner_directory(void) {
+  static const char* get_configuration_directory(void) {
```

Comment:
[This comment](https://github.com/tekezo/Karabiner-Elements/issues/112#issuecomment-256811473) made me realize that Karabiner-Elements as an application probably doesn't have access to shell variables.

There seems to be a way to set global shell variables, but it's not as convenient as a normal dotfile for commandline tools:
http://stackoverflow.com/a/588442

A simple compromise is to copy https://github.com/Hammerspoon/hammerspoon/pull/582 and default to `~/.config/karabiner-elements` unless a certain value is set in the Karabiner-Elements `.plist`. That doesn't scale to multiple applications the same way as `$XDG_CONFIG_HOME`, though. :-(